packages/query-core/src/query.ts

```diff
@@ -516,7 +515,12 @@ export class Query<
           return {
             ...state,
             fetchFailureCount: 0,
-            fetchMeta: action.meta ?? null,
+            fetchMeta: {
+              ...action.meta,
+              fetchedPages: action.meta?.fetchMore
+                ? state.fetchMeta?.fetchedPages
+                : 0,
```

Comment:
I think when a `fetch` action is triggered, we should always set `fetchPages` back to `0`. I see no case where we actually pass `fetchedPages` to the `fetch` action?